urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/yuccasrv/socketmngr/ServerSocketThread.java

```diff
@@ -0,0 +1,468 @@
+	    trustManagerFactory = TrustManagerFactory.getInstance( "SunX509" );
+	    trustManagerFactory.init( trustStore ); 
+	}
+
+
+	   
+	logger.log( Level.INFO,
+		    getClass().getName() + ".createSSLServerSocket(...)",
+		    "Creating key manager factory ... " );
+	KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance("SunX509");
+	keyManagerFactory.init( keyStore, (wrp_keyStorePass == null ? new char[0] : wrp_keyStorePass.getString().toCharArray()) );
+
+	
+
+	logger.log( Level.INFO,
+		    getClass().getName() + ".createSSLServerSocket(...)",
+		    "Creating SSL context (SSLv3) ... " );
+	SSLContext sslcontext = SSLContext.getInstance("SSLv3");
+	if( trustManagerFactory != null )  
+	    sslcontext.init( keyManagerFactory.getKeyManagers(), trustManagerFactory.getTrustManagers(), null);
+	else                               
+	    sslcontext.init( keyManagerFactory.getKeyManagers(), null, null);
+
+
+	logger.log( Level.INFO,
+		    getClass().getName() + ".createSSLServerSocket(...)",
+		    "Creating server socket factory ... " );
+	ServerSocketFactory ssf = sslcontext.getServerSocketFactory();
+
+
+	logger.log( Level.INFO,
+		    getClass().getName() + ".createSSLServerSocket(...)",
+		    "Creating server socket ... " );
+	SSLServerSocket serversocket = (SSLServerSocket)ssf.createServerSocket( bindPort );
+	return serversocket;
+	
+
+	//return null;
+
+    }
+    
+    /**
+     * This method fetches the 'value' entry from the SSL settings where the 'name' equals the
+     * passed key-name.
+     *
+     * If the key cannot be found, the method returns null.
+     **/
+    private BasicType getSSLConfigPropertyValue( Environment<String,BasicType> bindSettings,
+						 String keyName ) {
+
+	
+	List<Environment<String,BasicType>> children = bindSettings.getChildren( Constants.CONFIG_SERVER_LISTEN_PROPERTY );
+	Iterator<Environment<String,BasicType>> iter = children.iterator();
+	while( iter.hasNext() ) {
+
+	    Environment<String,BasicType> child = iter.next();
+	    BasicType property = child.get( Constants.CONFIG_SERVER_LISTEN_PROPERTY_NAME ); 
+	    if( property.getString().equalsIgnoreCase(keyName) ) {
+
+		//System.out.println( "--- SSL property '"+keyName+"' loaded: " + child.get("value") );
+		return child.get( Constants.CONFIG_SERVER_LISTEN_PROPERTY_VALUE );
+
+	    }
+
+	}
+	
+	return null;
+
+    }
+
+    private KeyStore loadKeyStore( String fileName,
+				   char[] password,
+				   String keyStoreType ) 
+	throws KeyStoreException,
+	       FileNotFoundException,
+	       IOException,
+	       NoSuchAlgorithmException,
+	       CertificateException {
+
+	if( keyStoreType == null )
+	    keyStoreType = "JKS";
+
+	/*this.logger.log( Level.INFO,
+			   getClass().getName() + ".createSSLServerSocket(...)",
+			   "Initializing key store '" + fileName + "' ... " );
+	*/
+	KeyStore keyStore = KeyStore.getInstance( keyStoreType );
+	keyStore.load( new FileInputStream(fileName), 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor: *FileInputStream*. Currently, there are execution paths that do not contain closure statements, for example, when KeyStore.load throws an exception. Ensure the object returned by *FileInputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).